controller/setting_controller.go

```diff
@@ -1078,31 +1078,53 @@ func (sc *SettingController) updateLogLevel(settingName types.SettingName) error
 }
 
 func (sc *SettingController) syncDefaultLonghornStaticStorageClass() error {
+	labels := types.GetDefaultLonghornStaticStorageClassLabels()
+
 	setting, err := sc.ds.GetSettingWithAutoFillingRO(types.SettingNameDefaultLonghornStaticStorageClass)
 	if err != nil {
 		return err
 	}
 
 	defaultStaticStorageClassName := setting.Value
 	_, err = sc.ds.GetStorageClassRO(defaultStaticStorageClassName)
-	if err != nil && apierrors.IsNotFound(err) {
-		allowVolumeExpansion := true
-		storageClass := &storagev1.StorageClass{
-			ObjectMeta: metav1.ObjectMeta{
-				Name: defaultStaticStorageClassName,
-			},
-			Provisioner:          types.LonghornDriverName,
-			AllowVolumeExpansion: &allowVolumeExpansion,
-			Parameters:           map[string]string{"staleReplicaTimeout": "30"},
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			allowVolumeExpansion := true
+			storageClass := &storagev1.StorageClass{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:   defaultStaticStorageClassName,
+					Labels: labels,
+				},
+				Provisioner:          types.LonghornDriverName,
+				AllowVolumeExpansion: &allowVolumeExpansion,
+				Parameters:           map[string]string{"staleReplicaTimeout": "30"},
+			}
+
+			if _, err := sc.ds.CreateStorageClass(storageClass); err != nil {
+				return err
+			}
+			return nil
 		}
+		return err
+	}
 
-		if _, err := sc.ds.CreateStorageClass(storageClass); err != nil {
-			return err
+	storageClassList, err := sc.ds.ListStorageClassesWithLabelsRO(labels)
+	if err != nil {
+		return err
+	}
+	for _, storageClass := range storageClassList {
+		if storageClass.Name == defaultStaticStorageClassName {
+			continue
+		}
+		if storageClass.DeletionTimestamp == nil {
+			sc.logger.Tracef("Deleting the old Longhorn static storage class %v", storageClass.Name)
+			if err := sc.ds.DeleteStorageClass(storageClass.Name); err != nil {
+				return err
+			}
 		}
-		return nil
 	}
```

Comment:
Shall we delete the storage class? User might just replace the static storage class with another one but doesn't want to delete it.

Reply:
Here only delete the storage class Longhorn creates (with the label `LonghornLabelManagedStorageClass = "managed-storage-class"`)

Reply:
Got it.